ts/src/gate.ts

```diff
@@ -6263,6 +6264,42 @@ export default class gate extends Exchange {
         return await this.privateFuturesPostSettleDualMode (this.extend (request, query));
     }
 
+    async fetchUnderlyingAssets (params = {}) {
```

Comment:
@Dan-krm I'm afraid this name might be too generic and confuse some users, fetchOptionUnderlyingAssets would make sense in your opinion?

Reply:
I went with fetchUnderlyingAssets instead of something more option specific because it can theoretically work with any type of derivative, the Okx version can work with options, futures and swap markets. Do you think including a non-required `type` argument would help clear some confusion?

Reply:
@Dan-krm yes I think we can do that for now